addons/website_sale/tests/test_website_sale_gmc.py

```diff
@@ -0,0 +1,701 @@
+        with self.mock_public_request(**kwargs):
+            self.items = self.products._get_gmc_items()
+        self.white_mouse_item = self.items[self.white_mouse]
+        self.black_mouse_item = self.items[self.black_mouse]
+
+
+@tagged('post_install', '-at_install')
+class TestWebsiteSaleGMCController(TestWebsiteSaleGMCCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.WebsiteSaleController = WebsiteSale()
+
+    def test_gmc_src_not_enabled(self):
+        """/gmc.xml route is accessible iff `website.enabled_gmc_src` is true"""
+        self.website.enabled_gmc_src = False
+        with self.assertRaises(NotFound), self.mock_public_request():
+            self.WebsiteSaleController.gmc_data_source()
+
+    def test_gmc_route(self):
+        """Test /gm.xml works under normal conditions"""
+        response = self.url_open('/gmc.xml')
+        self.assertEqual(200, response.status_code)
+        gmc_xml = etree.XML(response.content)  # valid xml
+
+        self.assertEqual(f'Home | {self.website.name}', gmc_xml.xpath('//title')[0].text)
+        self.assertURLEqual(
+            '/',
+            gmc_xml.xpath('//link')[0].text,
+        )
+        self.assertEqual(
+            'This is the homepage of the website',
+            gmc_xml.xpath('//description')[0].text,
+        )
+
+    def test_translation(self):
+        """Test translation of /gmc.xml content"""
+        fr_lang = self.env['res.lang']._activate_lang('fr_FR')
+        self.website.language_ids += fr_lang
+        self.website.name = 'The best website'
+        self.website.with_context(lang=fr_lang.code).name = 'Mon unique site'
+        self.white_mouse.with_context(lang=fr_lang.code).name = 'La souris magique blanche'
+        response = self.url_open('/fr/gmc.xml')
+        self.assertEqual(200, response.status_code)
+        gmc_xml = etree.XML(response.content)
+        self.assertEqual('Home | Mon unique site', gmc_xml.xpath('//title')[0].text)
+        self.assertURLEqual(
+            f'/{fr_lang.url_code}',
+            gmc_xml.xpath('//link')[0].text,
+            'The links must redirect to the french website.',
+        )
+        self.assertEqual(
+            'La souris magique blanche',
+            gmc_xml.xpath(
+                '//item[g:id="MAGIC-W"]/g:title', namespaces={'g': 'http://base.google.com/ns/1.0'}
+            )[0].text,
+        )
+
+    def test_pricelist(self):
+        """Test /gmc.xml with a specific pricelist"""
+        response = self.url_open('/gmc-eur%20christ.xml')
+        self.assertEqual(200, response.status_code)
+        gmc_xml = etree.XML(response.content)
+        self.assertEqual(
+            '86.9 EUR',
+            gmc_xml.xpath(
+                '//item[g:id="MAGIC-W"]/g:price', namespaces={'g': 'http://base.google.com/ns/1.0'}
+            )[0].text
+        )
+
+
+@tagged('post_install', '-at_install')
+class TestWebsiteSaleGMCItems(TestWebsiteSaleGMCCommon):
```

Comment:
Since there is no setup difference between this class and `TestWebsiteSaleGMCController`, I don't think it's necessary to use two separate classes :thinking: